test/unit/compilable/crlf.d

```diff
@@ -0,0 +1,79 @@
+module compilable.crlf;
+
+import support : afterEach, beforeEach, defaultImportPaths;
+
+@beforeEach initializeFrontend()
+{
+    import dmd.frontend : initDMD;
+    initDMD();
+}
+
+@afterEach deinitializeFrontend()
+{
+    import dmd.frontend : deinitializeDMD;
+    deinitializeDMD();
+}
+
+@("test CRLF and mixed line endings")
+unittest
+{
+    import std.algorithm : each, map;
+    import std.array : array, join;
+    import std.format : format;
+    import std.range : dropOne, dropBackOne;
+    import std.string : lineSplitter, chompPrefix;
+
+    import support : compiles;
+
+    enum crLFCode = q{
+        #!/usr/bin/env dmd -run
+
+        #line 4
+
+        void main()
+        {
+        }
+
+        // single-line comment
+
+        /*
+          multi-line comment
+        */
+
+        /+
+          nested comment
+        +/
+
+        /**
+          doc comment
+        */
+        void documentee() {}
+    }
+    .lineSplitter
+    .array
+    .dropBackOne
+    .dropOne
+    .map!(line => line.chompPrefix("        "))
```

Comment:
These range functions here are basically a quick hack to remove the extra indentation added with by the `q{` string literal syntax.

Reply:
You should probably replace the line endings with `\r\n` to match the original test as well.

Reply:
It does, one line below: https://github.com/dlang/dmd/pull/9849/files#diff-163b1fbf9b5db86d7742341305c7b422R57

Reply:
I wrote the code, both in this test and in the original test, to a file and diffed the result. No changes.

Reply:
Can convert this into a function named something along the lines of `trimPrefix(string prefix, string newline = "\n")` to make it about more self-documenting?

Reply:
Sure.

Reply:
Fixed.